flexmeasures/api/v3_0/assets.py

```diff
@@ -65,6 +65,42 @@
 partial_asset_schema = AssetSchema(partial=True, exclude=["account_id"])
 
 
+def format_json_field_change(field_name: str, old_value, new_value) -> str:
+    """Format JSON field changes using dictdiffer."""
+    try:
+        from dictdiffer import diff
+
+        if isinstance(old_value, list):
+            old_dict = {i: item for i, item in enumerate(old_value)}
+            new_dict = {i: item for i, item in enumerate(new_value)}
+        else:
+            old_dict, new_dict = old_value, new_value
+
+        if isinstance(old_dict, dict) and isinstance(new_dict, dict):
+            diff_results = list(diff(old_dict, new_dict))
+            changes = []
+            for change_type, key, value in diff_results:
+                if change_type == "change":
+                    changes.append(f"change {key}: {value[0]} -> {value[1]}")
+                elif change_type == "add":
+                    for item in value:
+                        changes.append(f"add {key}[{item[0]}]: {item[1]}")
```

Comment:
See my comment about "remove" - not sure why I'm seeing brackets in `add [site-peak-production]: 2kW`